### DISCUSSION TITLE
Bundling from string rather than from file

### DISCUSSION THREAD

**Discussion by leo-petrucci** [parcel-bundler/parcel#5208] [Q&A] [answered]
If I have a string of generated HTML that looks like this:
```
'<html>\n' +
      '  <body>\n' +
      '    <script src="/public/js/index.js"></script>\n' +
      '    <h1><%= pageName %></h1>\n' +
      '  </body>\n' +
      '</html>\n'
```
Is there any way for me to pass this to Parcel to generate an html file?

I tried creating a temp file and passing that and while that worked, it broke importing js as it's located in a completely different folder.

**Comment by mischnic** [comment, marked as answer]
> it broke importing js as it's located in a completely different folder.

How should Parcel know where to resolve relative urls from? So such a html snippet needs to have a path anyway for resolving paths.

**Reply by leo-petrucci** [reply]
Not sure, was hoping for some hacky way to tell it the root path myself. Thanks for clarifying this for me!

**Reply by mischnic** [reply]
With Parcel 2, you could invoke Parcel via the JS API (instead of the CLI) and write that file into the `MemoryFS` (similar to this: https://v2.parceljs.org/features/parcel-api/#outputting-to-an-in-memory-file-system)

**Reply by leo-petrucci** [reply]
Oh that's exactly what I'm looking for. I'm already using the js api and my solution was going to be creating files inside /tmp, but this seems like a much better option!

Thank you 😊

**Reply by leo-petrucci** [reply]
@mischnic How does the example work exactly? The whole code in the example doesn't seem to be working. Even having just `memoryFS` returns `memoryFS` is not a constructor.

**Reply by mischnic** [reply]
Here's another example: https://github.com/parcel-bundler/parcel/issues/4530#issuecomment-619398712. We should add that to the docs as well

**Reply by leo-petrucci** [reply]
Just to be clear, there's no way to tell Parcel my entry file is in the memory fs right? Parcel wants me to provide a string to my entry file (an html file at the moment) but there's no way for it to access the `fs` I created with a string.
